frame/system/src/lib.rs

```diff
@@ -486,6 +489,7 @@ decl_module! {
 			}
 
 			storage::unhashed::put_raw(well_known_keys::CODE, &code);
+			RuntimeUpgraded::put(true);
```

Comment:
technically this isn't the only way a runtime could be upgraded: any module can call `unhashed::set_storage(b":CODE", ...);` in order to change it. also, `set_storage(b":CODE", ...)` could be dispatched, which will also short-circuit an upgrade. i think allowing the first one to break things is fine (though should be included in the docs), but the second one should be prevented from setting the `:CODE` entry (or, if it does set it, then you should call `RuntimeUpgraded::put(true);`).

Reply:
Good catch. I have added a check which sets the value to true and also a test.